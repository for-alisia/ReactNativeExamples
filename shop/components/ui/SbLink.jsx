import React from 'react';
import { StyleSheet, View } from 'react-native';

// Components
import SbTouchable from './SbTouchable';
import SbText from './SbText';

// Theme
import theme from '../../theme';

const SbLink = (props) => {
  const { onPress, children, style } = props;
  return (
    <SbTouchable onPress={onPress}>
      <View style={{ ...styles.container, ...style }}>
        <SbText style={styles.link}>{children}</SbText>
        <View style={styles.line}></View>
      </View>
    </SbTouchable>
  );
};

const styles = StyleSheet.create({
  container: {
    alignSelf: 'center',
    padding: theme.padding.s,
  },
  link: {
    fontFamily: theme.fonts.montserratReg,
    color: theme.colors.primary,
  },
  line: {
    height: 1,
    backgroundColor: theme.colors.secondary,
  },
});

export default SbLink;
