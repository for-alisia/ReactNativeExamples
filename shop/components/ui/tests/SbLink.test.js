// @ts-nocheck
// Dependencies
import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';

// Component to test
import SbLink from '../SbLink';

describe('SbLink UI Component', () => {
  const testText = 'Hello';
  const mockOnPress = jest.fn();

  it('renders a link with a passed text', () => {
    const { getByText, toJSON } = render(<SbLink>{testText}</SbLink>);
    const text = getByText(testText);

    expect(text).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls a correct handler once', () => {
    const { getByText } = render(<SbLink onPress={mockOnPress}>{testText}</SbLink>);
    const link = getByText(testText);
    fireEvent.press(link);

    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });
});
