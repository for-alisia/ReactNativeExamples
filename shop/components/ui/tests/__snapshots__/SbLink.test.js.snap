// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SbLink UI Component renders a link with a passed text 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "padding": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#DC8080",
          "fontFamily": "montserrat-regular",
          "fontSize": 16,
        }
      }
    >
      Hello
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#D8C496",
          "height": 1,
        }
      }
    />
  </View>
</View>
`;
